src/utils/analytics.ts

```diff
@@ -0,0 +1,61 @@
+import { AnalyticsBrowser } from "@segment/analytics-next";
+import { ExtensionStorage, TempTransactionStorage } from "./storage";
+import { useState, useEffect } from "react";
+
+const analytics = AnalyticsBrowser.load({
+  writeKey: process.env.PLASMO_PUBLIC_SEGMENT_WRITEKEY
+});
+
+export enum EventType {
+  FUNDED = "FUNDED",
+  CONNECTED_APP = "CONNECTED_APP",
+  LOGIN = "LOGIN",
+  ONBOARDED = "ONBOARDED"
+}
+
+export const trackEvent = async (eventName: EventType, properties: any) => {
+  // first we check if we are allowed to collect data
+  const enabled = await ExtensionStorage.get<boolean>("setting_analytics");
```

Comment:
I moved this here, so we don't accidentally track anything if it is not allowed.